ferry.go

```diff
@@ -832,7 +839,7 @@ func (f *Ferry) ensureInitialized() {
 	// Note: the constructor shouldn't have a large amount of positional argument
 	//       so it is more readable.
 	//
-	if f.OverallState == "" {
+	if f.OverallState.Load() == "" {
```

Comment:
If this isn't set, it is still an interface, and should be compared against `nil`, right? 

```go
package main

import (
	"sync/atomic"
)

func main() {
	var m atomic.Value
	
	if m.Load() == "" {
		panic("not initialized")
	}
	
	if m.Load() == nil {
		println("m is nil")
	}
}

```
the above evaluates to:
```
m is nil
```

and does not panic.

Reply:
Good catch!

Reply:
> This is my bad probably for suggesting using `atomic.Value` 🤦 sorry Xiaoli

No worries. ❤️